### PR TITLE
Updating Geos to 3.7

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

This adds Geos ~~3.6~~ 3.7 support (necessary for GDAL because of a small API change from 3.5 to ~~3.6~~ 3.7).

3.7 was necessary for both Mac and Linux. 3.6 worked only for mac but Gdal for Linux requires Geos 3.7  

## How Has This Been Tested?

Running tests on Jenkins, waiting for them to finish. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: build.groovy [hunk 1]
```diff
@@ -83,17 +83,16 @@ node("${env.OS.toLowerCase()}") {
...
-              which conda
               conda config --set always_yes True
               conda config --set ssl_verify false 
-              conda create -n isis python=3
+              conda update -n base -c defaults conda
           """
            
           
           if (env.OS.toLowerCase() == "centos") {
               sh 'conda env update -n isis -f environment_gcc4.yml --prune'
```
Comment (krlberry) [inline build.groovy]: Doesn't this need to be changed to a `conda env create`, too, if the isis env isn't created on line 89?

**Review by krlberry — CHANGES_REQUESTED** [review]
1 minor thing. The rest looks good.

File: build.groovy (hunk 1 above)
Comment (Kelvinrr) [reply build.groovy]: yes, good catch. Been focusing on Mac too much.

**Comment by Kelvinrr** [comment]
Should be good for re-review.

**Review by krlberry — APPROVED** [review]
:+1: Looks good!

**Pull request merged**